### PR TITLE
update gateway service protobuf references

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Closes https://github.com/camunda/camunda-docs/issues/3367.

Updates protobuf references based on https://github.com/camunda/camunda/blob/86828ff3fd3caef48b3229c1f717466b35c955eb/gateway-protocol/src/main/proto/gateway.proto

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

**To backport upon approval**

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/apis-tools/zeebe-api/gateway-service.md [hunk 1]
```diff
@@ -160,8 +160,6 @@ message CancelProcessInstanceRequest {
   // the process instance key (as, for example, obtained from
   // CreateProcessInstanceResponse)
   int64 processInstanceKey = 1;
-  // a reference key chosen by the user and will be part of all records resulting from this operation
-  optional uint64 operationReference = 2;
```
Comment (akeller) [inline docs/apis-tools/zeebe-api/gateway-service.md:164]: I noticed these reference keys. Is there more context here? Is this not something we want to recommend or is there an alternative now?

Comment (christinaausley) [reply docs/apis-tools/zeebe-api/gateway-service.md:164]: These were removed in the protobuf, but I'll leave it up to @korthout if we want to keep this in the docs?

Comment (christinaausley) [reply docs/apis-tools/zeebe-api/gateway-service.md:164]: CC @megglos

Comment (christinaausley) [reply docs/apis-tools/zeebe-api/gateway-service.md:164]: @akeller As @megglos is out on parental leave for about a month, I think we can go ahead and merge this. If he returns and decides to add the reference keys we can certainly pop them back in!

**Review by akeller — APPROVED** [review]

**Pull request merged**